packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,264 @@
+import { ClickHouse, sql } from '../../operations/providers/clickhouse-client';
+import { SqlValue } from '../../operations/providers/sql';
+import { Emails, mjml } from '../../shared/providers/emails';
+import { Logger } from '../../shared/providers/logger';
+import { S3_CONFIG, type S3Config } from '../../shared/providers/s3-config';
+import { AuditLogClickhouseArrayModel, AuditLogType } from './audit-logs-types';
+
+@Injectable({
+  scope: Scope.Operation,
+})
+/**
+ * Responsible for accessing audit logs.
+ */
+export class AuditLogManager {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+    @Inject(S3_CONFIG) private s3Config: S3Config,
+    private emailProvider: Emails,
+    private session: Session,
+    private auth: AuthManager,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogManager' });
+  }
+
+  async getPaginatedAuditLogs(
+    organizationSlug: string,
+    pagination?: {
+      cursorId?: string | null;
+      cursorTimestamp?: string | null;
+      first: number | null;
+    },
+    filter?: { startDate?: Date; endDate?: Date },
+  ): Promise<{ data: AuditLogType[] }> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      throw new GraphQLError('Unauthorized: You are not authorized to perform this action');
+    }
+
+    this.logger.info(
+      'Getting audit logs (organizationId=%s, filter=%o, pagination=%o)',
+      organizationSlug,
+      filter,
+      pagination,
+    );
+    const newCursorId = pagination?.cursorId ? decodeHashBasedCursor(pagination?.cursorId) : null;
+    const lastYear = this.formatToClickhouseDateTime(
+      new Date(new Date().setFullYear(new Date().getFullYear() - 1)).toISOString(),
+    );
+    const limit = sql.raw(String(pagination?.first));
+    const cursorTimestamp = pagination?.cursorTimestamp
+      ? this.formatToClickhouseDateTime(pagination.cursorTimestamp)
+      : lastYear;
+
+    const where: SqlValue[] = [];
+    where.push(sql`organization_id = ${organizationSlug}`);
+
+    if (filter?.startDate && filter?.endDate) {
+      const from = this.formatToClickhouseDateTime(filter.startDate.toISOString());
+      const to = this.formatToClickhouseDateTime(filter.endDate.toISOString());
+      where.push(sql`timestamp >= ${from} AND timestamp <= ${to}`);
+    }
+    if (pagination?.cursorId && pagination?.cursorTimestamp) {
+      where.push(sql`id < ${String(newCursorId?.id)}`, sql`timestamp = ${cursorTimestamp}`);
+    } else if (pagination?.cursorId) {
+      where.push(sql`id < ${String(newCursorId?.id)}`);
+    } else if (pagination?.cursorTimestamp) {
+      where.push(sql`timestamp = ${cursorTimestamp}`);
+    } else {
+      where.push(sql`timestamp <= ${cursorTimestamp}`);
+    }
+
+    const whereClause = where.length > 0 ? sql`WHERE ${sql.join(where, ' AND ')}` : sql``;
+
+    const withoutLimit = sql`
+    SELECT *
```

Comment:
Please list the explicit columns here we are selecting. We avoid adding the "*" selector.